pkg/reconciler/knativeeventing/knativeeventing.go

```diff
@@ -90,14 +91,17 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, ke *v1alpha1.KnativeEven
 	ke.Status.InitializeConditions()
 	ke.Status.ObservedGeneration = ke.Generation
 
+	// TODO: I'm sure there's a more elegant way to do this
```

Comment:
Let's not do this at all, I'd say. In this case I'd suggest to make `CheckDeployments` returns a closure, like:

```golang
func CheckDeployments(kube kubernetes.Interface) Stage
  return func (ctx context.Context, manifest *mf.Manifest, instance v1alpha1.KComponent) {
    ...
  }
}
```

I'm assuming here that the clients aren't part of the reconcile context for a reason. If we'd want to have them on there, I'd rather discuss this on the framework in knative.dev/pkg and make it generally so.

Reply:
Interesting. That kinda baffles me, given [this definition](https://book.kubebuilder.io/cronjob-tutorial/controller-overview.html)...

"It’s a controller’s job to ensure that, for any given object, the actual state of the world (both the cluster state, and potentially external state like running containers for Kubelet or loadbalancers for a cloud provider) matches the desired state in the object."

How else would the controller reconcile the state of the cluster without clients? Right now, the only way our operator does anything is through the manifest's embedded client. I could use that, of course. That would simplify the API considerably -- I wouldn't need the closure since the manifest is already passed. Actually, one could argue some of our awkward "append to the root manifest member" logic is only due to the fact that the client isn't available in the reconciler's context.

I would love to know the reasoning behind why clients aren't available to the reconciler.

Reply:
Client's are absolutely available to the reconciler. They just are not available to the **context** of each reconcile operation. I don't think that contradicts this definition. TBH, I don't know the reasoning behind not making them available on the reconcile context but it probably has to do with burying client gets too deeply in potential APIs like this one. It gets unwieldy over time.

Reply:
But aren't they only available if the reconciler author takes care to create a member to hold it? That seems like something the framework could better support maybe?

Reply:
Maybe, hence I'd want to discuss this in /pkg for the entire framework and not go a "hacky" way here on our end if we can help it.